common/lib/capa/capa/responsetypes.py

```diff
@@ -363,15 +368,23 @@ def make_hint_div(self, hint_node, correct, student_answer, question_tag,
         self.capa_module.runtime.track_function('edx.problem.hint.feedback_displayed', event_info)
 
         # Form the div-wrapped hint texts
-        hints_wrap = u''.join(
-            [u'<div class="{0}">{1}</div>'.format(QUESTION_HINT_TEXT_STYLE, dct.get('text'))
-             for dct in hint_log]
+        hints_wrap = HTML('').join(
+            [HTML('<div class="{question_hint_text_style}">{hint_content}</div>').format(
+                question_hint_text_style=QUESTION_HINT_TEXT_STYLE,
+                hint_content=HTML(dct.get('text'))
```

Comment:
Unless you have already determined this, we need to figure out if HTML is a legitimate value for hint "text". If it should be plain text, it should be wrapped with Text instead of HTML.